staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto

```diff
@@ -199,7 +199,11 @@ message PortMapping {
 }
 
 enum MountPropagation {
-    // No mount propagation ("private" in Linux terminology).
+    // No mount propagation ("rprivate" in Linux terminology).
+    // However, an implementation may choose "rslave" instead when "rprivate" is not
+    // applicable.
+    // cri-dockerd (Docker) is known to choose "rslave" when the mount source contains
```

Comment:
I wonder if other runtimes will have the same problem if attempt to mount their root directories or some other relevant directories that are used to run container. I wonder if this comment can be generalized once fixed for other runtimes...

@mikebrow @mrunalp

Reply:
I'd argue that cri-dockerd's behavior is wrong (but unfixable for compatibility sake.  It should have just raised an explicit error when the requested propagation ("None" in the yaml) is not applicable, as falling back to "HostToContainer" (aka "rslave")  may result in leaking unexpected data to the container, etc.

Reply:
Updated the PR to clarify the expected behavior